driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionProseTestSpecification.groovy

```diff
@@ -19,490 +19,159 @@ package com.mongodb.client
+                .keyId(localDataKeyId))
 
         then:
-        e = thrown(MongoClientException)
-        e.getCause() instanceof MongoCryptException
-    }
-
-    // TODO: enable this
-    @Ignore
-    def 'should fail to encrypt invalid values with randomized encryption'() {
-        given:
-        dataKeyCollection.insertOne(localDataKeyDocument)
-        def keyVault = ClientEncryptions.create(ClientEncryptionSettings.builder()
-                .keyVaultMongoClientSettings(getMongoClientSettings())
-                .keyVaultNamespace(keyVaultNamespace.fullName)
-                .kmsProviders(localProviderProperties)
-                .build())
-        def options = new EncryptOptions('AEAD_AES_256_CBC_HMAC_SHA_512-Randomized')
-                .keyId(localDataKeyDocument.getBinary('_id'))
+        localEncrypted.asBinary().getType() == (byte) 6
 
         when:
-        keyVault.encrypt(new BsonBinary((byte) 6,
-                Base64.decoder.decode('AQAAAAAAAAAAAAAAAAAAAAACN0NwWlfe6YPGDEw+ObxEzbEtk45ewF3sIH2Oj7F0xd3GYoxCGCIp9gg0Q1uHTwdVWwG58SFhJyo4305IVoikEQ==')),
-                options)
+        autoEncryptingDataCollection.insertOne(new BsonDocument('_id', new BsonString('local'))
+                .append('value', localEncrypted))
 
         then:
-        thrown(MongoCryptException)
-    }
-
-    def 'should encrypt values with randomized encryption that are invalid with deterministic encryption'() {
-        given:
-        dataKeyCollection.insertOne(localDataKeyDocument)
-        def keyVault = ClientEncryptions.create(ClientEncryptionSettings.builder()
-                .keyVaultMongoClientSettings(getMongoClientSettings())
-                .keyVaultNamespace(keyVaultNamespace.fullName)
-                .kmsProviders(localProviderProperties)
-                .build())
-        def options = new EncryptOptions('AEAD_AES_256_CBC_HMAC_SHA_512-Random')
-                .keyId(localDataKeyDocument.getBinary('_id'))
+        autoEncryptingDataCollection.find(eq('_id', new BsonString('local'))).first().getString('value')
+                .value == 'hello local'
 
         when:
-        keyVault.encrypt(new BsonDouble(1.0), options)
+        def localEncryptedWithAltName = clientEncryption.encrypt(new BsonString('hello local'),
+                new EncryptOptions('AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic')
+                .keyAltName('local_altname'))
 
         then:
-        noExceptionThrown()
+        localEncryptedWithAltName == localEncrypted
 
         when:
-        keyVault.encrypt(new BsonDecimal128(Decimal128.parse('1.0')), options)
+        def awsDataKeyId = clientEncryption.createDataKey('aws',
+                new DataKeyOptions().keyAltNames(['aws_altname'])
+                .masterKey(new BsonDocument('region', new BsonString('us-east-1'))
+                .append('key', new BsonString('arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0'))))
```

Comment:
Another omission from the spec test itself. I did not mention the region or CMK.